graphium/fingerprinting/data.py

```diff
@@ -0,0 +1,216 @@
+            self.splits = list(self.smiles.keys())
+        
+        else:
+            # Check which splits are needed
+            self.splits = []
+            add_all = self.benchmark is not None or self.splits_path is not None
+            if add_all or self.split_val + self.split_test < 1:
+                self.splits.append("train")
+            if add_all or self.split_val > 0:
+                self.splits.append("valid")
+            if add_all or self.split_test > 0:
+                self.splits.append("test")
+
+            self.data = {
+                split: {
+                    "smiles": [],
+                    "labels": [],
+                    "fps": {},
+                }
+                for split in self.splits
+            }
+
+            for model, layers in self.pretrained_models.items():
+                predictor = PredictorModule.load_pretrained_model(model, device=self.device)
+                predictor.featurization.pop("max_num_atoms", None)
+
+                # Featurization
+                if self.benchmark is None:
+                    assert self.df_path is not None, "df_path must be provided if not using an integrated benchmark"
+
+                    # Add a dummy task column (filled with NaN values) in case no such column is provided
+                    smiles_df = pd.read_csv(self.df_path)
+                    task_cols = [col for col in smiles_df if col.startswith("task_")]
+                    if len(task_cols) == 0:
+                        df_path = ".".join(self.df_path.split(".")[:-1])
+                        smiles_df["task_dummy"] = np.nan
+                        smiles_df.to_csv(f"{df_path}_with_dummy_task_col.csv", index=False)
+                        self.df_path = f"{df_path}_with_dummy_task_col.csv"
+
+                    task_specific_args = {
+                        "fingerprinting": {
+                            "df_path": self.df_path,
+                            "smiles_col": "smiles",
+                            "label_cols": "task_*",
+                            "task_level": "graph",
+                            "splits_path": self.splits_path,
+                            "split_type": self.split_type,
+                            "split_val": self.split_val,
+                            "split_test": self.split_test,
+                            "seed": self.data_seed,
+                        }
+                    }
+                    label_key = "graph_fingerprinting"
+
+                    datamodule = MultitaskFromSmilesDataModule(
+                        task_specific_args=task_specific_args,
+                        batch_size_inference=128,
+                        featurization=predictor.featurization,
+                        featurization_n_jobs=0,
+                        processed_graph_data_path=f"{self.mol_cache_dir}/mols/",
+                    )
+
+                elif self.benchmark == "tdc":
+                    datamodule = ADMETBenchmarkDataModule(
```

Comment:
Perhaps change the name of that class to `TDCBenchmarkDatamodule`? Or is it used for Polaris too? Then `PolarisTDCBenchmarkDatamodule`?

Reply:
Changed to `TDCBenchmarkDatamodule`.